# Enforce consistent indentation in html template (html-indent)

Please describe the origin of the rule here.

## :book: Rule Details

This rule aims to...

Examples of **incorrect** code for this rule:

```html
<template>
    <div>
    </div>
</template>
```

Examples of **correct** code for this rule:

```html
<template>
    <div>
    </div>
</template>
```

## :wrench: Options

This rule has a mixed option:

For example, for 2-space indentation:

```
vue/html-indent: [2, 2]
```

Or for tabbed indentation:

```
vue/html-indent: [2, 'tab']
```
