---
pageClass: rule-details
sidebarDepth: 0
title: vue/valid-component-name
description: enforce consistency in component names
---

# vue/valid-component-name

> enforce consistency in component names

- :exclamation: <badge text="This rule has not been released yet." vertical="middle" type="error"> _**This rule has not been released yet.**_ </badge>

## :book: Rule Details

This rule ....

<eslint-code-block :rules="{'vue/valid-component-name': ['error']}">

```vue
<template>

</template>
```

</eslint-code-block>

## :wrench: Options

Nothing.

## :mag: Implementation

- [Rule source](https://github.com/vuejs/eslint-plugin-vue/blob/master/lib/rules/valid-component-name.js)
- [Test source](https://github.com/vuejs/eslint-plugin-vue/blob/master/tests/lib/rules/valid-component-name.js)
